sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -378,11 +378,21 @@ object GpuCast extends Arm {
       // ansi cast from larger-than-long integral-like types, to long
       case (dt: DecimalType, LongType) if ansiMode =>
         // This is a work around for https://github.com/rapidsai/cudf/issues/9282
-        val min = BigDecimal(Long.MinValue)
+        val bigDecimalMin = BigDecimal(Long.MinValue)
             .setScale(dt.scale, BigDecimal.RoundingMode.DOWN).bigDecimal
-        val max = BigDecimal(Long.MaxValue)
+        val bigDecimalMax = BigDecimal(Long.MaxValue)
```

Comment:
bigDecimalMax/Min are not minimums of BigDecimal,  the names are confusing. The conversion from long is probably non-trivial, might be worthwhile to make these class vals

Reply:
The `BigDecimal` will have to be re-scaled and most of the calculation is done in that method. I am not sure what we will gain by just making a `BigDecimal(Long.MIN)` as a class val. I can still do it but I just wanted to make sure I was understanding you correctly.

Reply:
Are we gaining anything by keeping recomputing the constants like `BigDecimal(Long.MaxValue)`?

If nothing else it's a bunch of unnecessary object allocations

https://github.com/scala/scala/blob/2.12.x/src/library/scala/math/BigDecimal.scala#L211
https://github.com/frohoff/jdk8u-dev-jdk/blob/da0da73ab82ed714dc5be94acd2f0d00fbdfe2e9/src/share/classes/java/math/BigDecimal.java#L1217-L1223
https://github.com/frohoff/jdk8u-dev-jdk/blob/da0da73ab82ed714dc5be94acd2f0d00fbdfe2e9/src/share/classes/java/math/BigDecimal.java#L1217-L1223

Reply:
Note `BigDecimal` instances are just like `Integer` immutable. So `setScale` will produce another object

Reply:
It is a valid argument depending on how frequent casting Decimals to Longs is done in a workload. The other concern is using static/constants increases the footprint of the VM and slows down the initialization.
Anyway, if this is going to be addressed within the same PR, then I suggest to create constants in a `util` to be used in different classes.
The repo has two other locations that use constants BigDecimal.
in arithemtic.scala there are two local variables 
```
 val zero = BigDecimal(0).bigDecimal
```
Then we can create three constants `BigDecimal(0).bigDecimal`, BigDecimal(Long.MaxValue), and BigDecimal(Long.MinValue)

All other constant BigDecimals are in test classes which we can ignore.

Reply:
Can we please handle creating of the util class in a separate PR?

Reply:
I agree it is fair. Both `BigDecimal(Long.MaxValue)`, and `BigDecimal(Long.MinValue)` were not introduced in this PR.

Reply:
I think we don't have yet to worry about the constant pool because of the two constants being added. We can do a more sweeping refactoring in a separate PR.